src/odemis/acq/stitching/_tiledacq.py

```diff
@@ -413,22 +434,105 @@ def save_tile(ix, iy, das):
         # Run in a separate thread
         threading.Thread(target=save_tile, args=(ix, iy, das), ).start()
 
-    def _acquireTile(self, i, ix, iy):
+    def _constructCube(self, images):
+        """
+        Construct xyz cube from a  z stack of images
+        :param images: (list) list of z ordered images
+        :return: (DataArray) the da of the xyz cube
+        """
+        # images is a list of 3 dim data arrays.
+        ret = []
+        for image in images:
+            stack = numpy.dstack(image)
+            stack = numpy.swapaxes(stack, 1, 2)
+            ret.append(stack[0])
```

Comment:
I think this is an extremely convoluted way to do that. I know this code is not from you, but from the zstack plugin, but now that see it, it's scary! I think that something like this should work (untested!):
```python
# Will fail on purpose if the images contain more than 2 dimensions
ret = numpy.array([im.reshape(im.shape[-2:]) for im in images])
```